portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,35 +30,44 @@
 import javax.servlet.http.HttpServletResponse;
 
 /**
- * Locally stores a List of Comment Strings which can be Read on GET or Appended to on POST
+ * Uses Datastore to store Comment Objects which can be Read on GET or Added to on POST
  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-  private static final Gson gson = new Gson();
-  private static final List<String> comments = new ArrayList<>();
+  private static Gson gson = new Gson();
 
   /**
-   * Gives response containing a single Json List with all stored comments
+   * Gives response containing a single Json List with all stored comments from datastore
+   * List sorted from oldest to newest comment
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      comments.add((String) entity.getProperty("text"));
+    }
     String json = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
   /**
-   * Iff request has a non-null comment-text parameter, store the parameter's string in the local comment list
+   * Iff request has a non-null comment-text parameter,
+   *   then create and store the parameter's String as a Datastore Comment Object
    * Always redirect to index.html
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("comment-text");
     if(comment != null){
-      comments.add(comment);
+      long timestamp = System.currentTimeMillis();
       Entity commentEn = new Entity("Comment");
       commentEn.setProperty("text",comment);
+      commentEn.setProperty("timestamp",timestamp);
```

Comment:
timestampMs or timestamp_ms?

Reply:
I'm using timestampMs now replaces timestamp for the long name and attribute name.